var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -96,20 +83,20 @@ class Dealii(CMakePackage):
 
     # optional dependencies
     depends_on("mpi",              when="+mpi")
-    depends_on("arpack-ng+mpi",    when='+arpack+mpi')
+    depends_on("arpack-ng",        when='+arpack')
     depends_on("doxygen+graphviz", when='+doc')
     depends_on("graphviz",         when='+doc')
     depends_on("gsl",              when='@8.5.0:+gsl')
-    depends_on("hdf5+mpi",         when='+hdf5+mpi')
+    depends_on("hdf5",             when='+hdf5')
     depends_on("metis@5:",         when='+metis')
-    depends_on("netcdf+mpi",       when="+netcdf+mpi")
-    depends_on("netcdf-cxx",       when='+netcdf+mpi')
+    depends_on("netcdf",           when="+netcdf")
+    depends_on("netcdf-cxx",       when='+netcdf')
     depends_on("oce",              when='+oce')
-    depends_on("p4est",            when='+p4est+mpi')
```

Comment:
this is for example wrong, `p4est` builds only with `mpi`, so in this instance `deal.ii` can use `p4est` if and only if it is build with `mpi` itself. I would certainly recommend splitting this PR into at least

1. `mpi=True` by default
2. the rest...

So that one could take a closer look. I am afraid you may be breaking more things that you try to fix here.